src/wallet/rpcwallet.cpp

```diff
@@ -2977,6 +3121,126 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     return ValueFromAmount(nBalance);
 }
 
+UniValue z_getbalanceforaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "z_getbalanceforaddress \"address\" ( minconf )\n"
+            "\nReturns the per-protocol (pool) balances of a Unified Address belonging to the node's wallet."
+            "\nCAUTION: If the wallet has only an incoming viewing key for this address, spends cannot be"
+            "\ndetected, and so the returned balance may be larger than the actual balance.\n"
+            "\nArguments:\n"
```

Comment:
I really wonder to what degree we should be creating new APIs that focus on addresses, rather than accounts. In general, "x for address" doesn't work great in the presence of diversified addresses anyway; the amounts received by a specific diversified address doesn't matter as much as the amount of spend authority associated with the keys from which those diversified addresses were derived.

Reply:
@str4d do you have an opinion here?

Reply:
The only addresses for which we would only have the incoming viewing key, are Sprout addresses tracked via an imported viewing key. So we could just prevent those from being used here, and require using `z_getbalance` for those (treating it all as deprecated).